SCRAM/createSymLinks.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+src=$1            #; shift  #source directory to search directories in
+des=$2            #; shift  #destination directory to create symlinks in
+depth=$3          #; shift  #how deep in source directory we search
+subdir=$4 || ""   #; shift  #sub-directory to search. There is a special cases
+                  #         # . mean find directory with same name as parent e.g. in LCG project we have PackageA/PackageA
+linkdir=$5 || ""  #; shift  #name of symlink to creat
+srcfilter=""
+srcnfilter=""
+if [ $# -gt 5 ]; then
+  shift 5
+  while [ $# -gt 0 ] ; do
+    arg="$(echo "$1" | sed 's/^\s+//')"
+    arg="$(echo "$1" | sed 's/\s*$//')"
+    if [[ $arg =~ ^-(.+)$ ]]; then
+      srcnfilter="${srcnfilter}${BASH_REMATCH[1]}|"
+    elif [[ $arg =~ ^(\+|)(.+)$ ]]; then
+      srcfilter="${srcfilter}${BASH_REMATCH[2]}|"
+    fi
+    shift
+  done
+fi
+srcfilter="$(echo "$srcfilter" | sed 's/\|$//')"
+srcnfilter="$(echo "$srcnfilter" | sed 's/\|$//')"
+
+function getSubDir () {
+  local dir=$1
+  local sdir=$2
+  if [ "X$sdir" = "X" ] ; then printf "\n" && return; fi
+  if [ "X$sdir" = "X." ] ; then sdir=$"/`basename $dir`" ; else sdir="/${sdir}" ; fi
```

Comment:
typo here. change `sdir=$"/`basename $dir`"` to `sdir=$"/`basename $sdir`"`

Reply:
> typo here. change `sdir=$"/`basename $dir`"` to `sdir=$"/`basename $sdir`"`

Perl version:
  if ($sdir eq "."){$sdir="/".basename($dir);}
  else{$sdir="/${sdir}";}

Should it be: basename $sdir or leave as in perl version?

Reply:
you are right, I did not see the `dir` variable at line number 28